nipype/interfaces/base/traits_extension.py

```diff
@@ -465,3 +471,120 @@ class InputMultiObject(MultiObject):
 
 InputMultiPath = InputMultiObject
 OutputMultiPath = OutputMultiObject
+
+
+class Tuple(traits.BaseTuple):
+    """Defines a new type of Tuple trait that reports inner types."""
+
+    def init_fast_validator(self, *args):
+        """Set up the C-level fast validator."""
+        super(Tuple, self).init_fast_validator(*args)
+        self.fast_validate = args
+
+    def inner_traits(self):
+        """Return the *inner trait* (or traits) for this trait."""
+        return self.types
+
+
+class PatchedEither(TraitType):
+    """Defines a trait whose value can be any of of a specified list of traits."""
+
+    def __init__(self, *traits, **metadata):
+        """Create a trait whose value can be any of of a specified list of traits."""
+        metadata['alternatives'] = tuple(trait_from(t) for t in traits)
+        self.trait_maker = _TraitMaker(
+            metadata.pop("default", None), *traits, **metadata)
+
+    def as_ctrait(self):
+        """Return a CTrait corresponding to the trait defined by this class."""
+        return self.trait_maker.as_ctrait()
+
+
+traits.Tuple = Tuple
+traits.Either = PatchedEither
+
+
+def _rebase_path(value, cwd):
+    if isinstance(value, list):
+        return [_rebase_path(v, cwd) for v in value]
+
+    try:
+        value = Path(value)
+    except TypeError:
+        pass
+    else:
+        try:
+            value = Path(value).relative_to(cwd)
```

Comment:
Already coerced in L512:

```suggestion
            value = value.relative_to(cwd)
```